bin/tx-prover/src/utils.rs

```diff
@@ -1,20 +1,71 @@
 use std::time::Duration;
 
+use opentelemetry::{trace::TracerProvider as _, KeyValue};
+use opentelemetry_sdk::{
+    runtime,
+    trace::{RandomIdGenerator, Sampler, TracerProvider},
+    Resource,
+};
+use opentelemetry_semantic_conventions::{
+    resource::{SERVICE_NAME, SERVICE_VERSION},
+    SCHEMA_URL,
+};
 use pingora::{http::ResponseHeader, Error, ErrorType};
 use pingora_proxy::Session;
 use tonic::transport::Channel;
 use tonic_health::pb::health_client::HealthClient;
+use tracing::Level;
+use tracing_subscriber::{layer::SubscriberExt, Registry};
+
+pub const TRACING_TARGET_NAME: &str = "miden-tx-prover";
 
 const RESOURCE_EXHAUSTED_CODE: u16 = 8;
 
-pub(crate) fn setup_tracing() {
-    // Set a default log level if `RUST_LOG` is not set
-    if std::env::var("RUST_LOG").is_err() {
-        std::env::set_var("RUST_LOG", "info"); // Default to 'info' level
-    }
-    tracing_subscriber::fmt::init();
+// Construct TracerProvider for OpenTelemetryLayer
+pub(crate) fn init_tracer_provider() -> TracerProvider {
+    let exporter = create_span_exporter();
+
+    TracerProvider::builder()
+        .with_sampler(Sampler::ParentBased(Box::new(Sampler::TraceIdRatioBased(1.0))))
```

Comment:
Not necessarily for this PR: Not sure if this will impact performance on production since it will depends on traces, but maybe this `1.0` could be an env var or configurable somehow.

Reply:
That would be great. It might be good to determine if the tracing/logging/metrics related configuration should be done through env vars like `RUST_LOG=<level>` or use a configuration file.

Reply:
Why not just make it a cli arg with env var support. clap supports this out the box, and personally I really dislike having these config files 😅 They just make deployment a nightmare.

Reply:
Opened up [this issue](https://github.com/0xPolygonMiden/miden-base/issues/1026) to tackle this.